qutebrowser/utils/version.py

```diff
@@ -249,42 +249,121 @@ def _release_info() -> Sequence[Tuple[str, str]]:
+
+        for attribute_name in self._version_attributes:
+            if hasattr(module, attribute_name):
+                version = getattr(module, attribute_name)
+                assert isinstance(version, (str, float))
+                self._version = str(version)
+                break
+
+    def get_version(self) -> Optional[str]:
+        """Finds the module version if it exists."""
+        if not self._initialized:
+            self._initialize_info()
+        return self._version
+
+    def is_installed(self) -> bool:
+        """Checks whether the module is installed."""
+        if not self._initialized:
+            self._initialize_info()
+        return self._installed
+
+    def is_outdated(self) -> Optional[bool]:
+        """Checks whether the module is outdated.
+
+        Return:
+            A boolean when the version and minimum version are both defined.
+            Otherwise `None`.
+        """
+        version = self.get_version()
+        if (
+            not self.is_installed()
+            or version is None
+            or self.min_version is None
+        ):
+            return None
+        return version < self.min_version
+
+
+MODULE_INFO: Mapping[str, ModuleInfo] = collections.OrderedDict([
+    (name, ModuleInfo(name, version_attributes, min_version))
+    for (name, version_attributes, min_version) in
+    (
+        ('sip', ('SIP_VERSION_STR'), None),
+        ('colorama', ('VERSION', '__version__'), None),
+        ('pypeg2', ('__version__',), None),
+        ('jinja2', ('__version__',), None),
+        ('pygments', ('__version__',), None),
+        ('yaml', ('__version__',), None),
+        ('adblock', ('__version__',), "0.3.2"),
+        ('cssutils', ('__version__',), None),
+        ('attr', ('__version__',), None),
+        ('PyQt5.QtWebEngineWidgets', (), None),
+        ('PyQt5.QtWebEngine', ('PYQT_WEBENGINE_VERSION_STR',), None),
+        ('PyQt5.QtWebKitWidgets', (), None),
+    )
+])
+
+
 def _module_versions() -> Sequence[str]:
     """Get versions of optional modules.
 
     Return:
         A list of lines with version info.
     """
     lines = []
-    modules: Mapping[str, Sequence[str]] = collections.OrderedDict([
-        ('sip', ['SIP_VERSION_STR']),
-        ('colorama', ['VERSION', '__version__']),
-        ('pypeg2', ['__version__']),
-        ('jinja2', ['__version__']),
-        ('pygments', ['__version__']),
-        ('yaml', ['__version__']),
-        ('adblock', ['__version__']),
-        ('cssutils', ['__version__']),
-        ('attr', ['__version__']),
-        ('PyQt5.QtWebEngineWidgets', []),
-        ('PyQt5.QtWebEngine', ['PYQT_WEBENGINE_VERSION_STR']),
-        ('PyQt5.QtWebKitWidgets', []),
-    ])
-    for modname, attributes in modules.items():
-        try:
-            module = importlib.import_module(modname)
-        except (ImportError, ValueError):
-            text = '{}: no'.format(modname)
+    for mod_info in MODULE_INFO.values():
```

Comment:
Maybe add a `ModuleInfo.__str__` generating the appropriate string? Then `_module_versions` could just be something like:

```python3
return [str(mod_info) for mod_info in MODULE_INFO.values()]
```